### PR TITLE
Add support for JSON output format

### LINKED ISSUE #1506: Structured Logging for Terragrunt Logs
Hello all! Currently Terragrunt logs output in an unstructured format, for example:
```
[terragrunt] [/Users/josue.valverde/Code/super-cool-terragrunt-project] 2021/01/21 12:39:36 Generated file /Users/josue.valverde/Code/super-cool-terragrunt-project/.terragrunt-cache/12YoLO/bAggiNS42/super-cool-terragrunt-project/backend.tf.
```

To improve readability and log searching through external log aggregators or monitoring systems like DataDog and Honeycomb.io, would it be possible to structure these? For example, having the above structured to be:
```
{
  "runner" : "terragrunt",
  "datetime" : "2021/01/21 12:39:36",
  "action" : "Generated file",
  "message" : "/Users/josue.valverde/Code/super-cool-terragrunt-project/.terragrunt-cache/12YoLO/bAggiNS42/super-cool-terragrunt-project/backend.tf."
}
```

I'm not sure how easy it'd be to split the logs by attributes, but it would make it so much easier to read these logs even while running in the CLI, as it would be clearer what is happening with lines of JSON-formatted logs:
```
{...}
{...}
{...}
```

Thanks!

Somewhat related logging feature: https://github.com/gruntwork-io/terragrunt/issues/662

### PR DESCRIPTION
**Pull request by denis256**
## Description

Included changes:
* Added new CLI flags :
  - `terragrunt-json-log` wrap Terragrunt output in JSON format
  - `terragrunt-tf-logs-to-json` wrap Terraform output in JSON messages, works when `terragrunt-json-log` is passed too
* Added documentation pages
* Added tests

Fixes #1506.

Example output:
https://gist.github.com/denis256/f0f8c076c67d65a3e2a425b6eff0a5c8

Run-all output:

https://gist.github.com/denis256/eeead6221b18d901630898c7d39db398

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

Added CLI flags to output logs in JSON format.

### CODE REVIEW

File: cli/commands/flags.go [hunk 1]
```diff
@@ -192,6 +194,18 @@ func NewGlobalFlags(opts *options.TerragruntOptions) cli.Flags {
 			EnvVar:      "TERRAGRUNT_NO_COLOR",
 			Usage:       "If specified, Terragrunt output won't contain any color.",
 		},
+		&cli.BoolFlag{
+			Name:        FlagNameTerragruntJsonLog,
+			Destination: &opts.JsonLogFormat,
+			EnvVar:      "TERRAGRUNT_JSON_LOG",
+			Usage:       "If specified, Terragrunt will outout logs in JSON format.",
```
Comment (brikis98) [inline cli/commands/flags.go:201]:
```suggestion
			Usage:       "If specified, Terragrunt will output its logs in JSON format.",
```

File: cli/commands/flags.go [hunk 2]
```diff
@@ -192,6 +194,18 @@ func NewGlobalFlags(opts *options.TerragruntOptions) cli.Flags {
...
+		},
+		&cli.BoolFlag{
+			Name:        FlagNameTerragruntTfLogJson,
+			Destination: &opts.TerraformLogsToJson,
+			EnvVar:      "TERRAGRUNT_TF_JSON_LOG",
+			Usage:       "If specified, Terragrunt will wrap Terraform output in JSON.",
```
Comment (brikis98) [inline cli/commands/flags.go:207]: Is it wrapping both Terraform `stdout` and `stderr`? Or just `stderr`?

```suggestion
			Usage:       "If specified, Terragrunt will wrap Terraform stdout and stderr in JSON.",
```

File: docs/_docs/04_reference/cli-options.md [hunk 3]
```diff
@@ -1032,3 +1034,17 @@ When this flag is set, Terragrunt does not update the remote state bucket, which
 **Environment Variable**: `TERRAGRUNT_DISABLE_COMMAND_VALIDATION` (set to `true`)
 
 When this flag is set, Terragrunt will not validate the terraform command, which can be useful when need to use non-existent commands in hooks.
+
+### terragrunt-json-log
+
+**CLI Arg**: `--terragrunt-json-log`
+**Environment Variable**: `TERRAGRUNT_JSON_LOG` (set to `true`)
+
+When this flag is set, Terragrunt will output logs in JSON format.
```
Comment (brikis98) [inline docs/_docs/04_reference/cli-options.md:1043]:
```suggestion
When this flag is set, Terragrunt will output its logs in JSON format.
```

File: docs/_docs/04_reference/cli-options.md [hunk 4]
```diff
@@ -1032,3 +1034,17 @@ When this flag is set, Terragrunt does not update the remote state bucket, which
...
+
+### terragrunt-tf-logs-to-json
+
+**CLI Arg**: `--terragrunt-tf-logs-to-json`
+**Environment Variable**: `TERRAGRUNT_TF_JSON_LOG` (set to `true`)
+
+When this flag is set, Terragrunt will wrap Terraform output in JSON log messages, works only with `--terragrunt-json-log` flag.
```
Comment (brikis98) [inline docs/_docs/04_reference/cli-options.md:1050]:
```suggestion
When this flag is set, Terragrunt will wrap Terraform `stdout` and `stderr` in JSON log messages. Works only with `--terragrunt-json-log` flag.
```

**Review by brikis98 — COMMENTED** [review]
Thanks!

**Comment by denis256** [comment]
Added Run-all output:
https://gist.github.com/denis256/eeead6221b18d901630898c7d39db398

Output also has "workingDir" field with path to module
```
{"level":"info","msg":"  + resource \"local_file\" \"file\" {","time":"2024-01-11T21:31:47Z","workingDir":"/projects/gruntwork/terragrunt-tests/run-all/app2"}

```

**Comment by denis256** [comment]
Added to JSON output field `executedCommandArgs` containing arguments passed to terraform

Example output:
```
{"executedCommandArgs":["--version"],"level":"info","msg":"Your version of Terraform is out of date! The latest version","time":"2024-01-16T15:53:35Z","workingDir":"/projects/gruntwork/terragrunt-tests/run-all/app1"}
{"executedCommandArgs":["--version"],"level":"info","msg":"is 1.6.6. You can update by downloading from https://www.terraform.io/downloads.html","time":"2024-01-16T15:53:35Z","workingDir":"/projects/gruntwork/terragrunt-tests/run-all/app1"}
{"executedCommandArgs":["apply","-auto-approve","-input=false"],"level":"info","msg":"local_file.file: Refreshing state... [id=db8a09739ef011f0b98fede9fea846ab90ea6e4d]","time":"2024-01-16T15:53:35Z","workingDir":"/projects/gruntwork/terragrunt-tests/run-all/app3"}
{"executedCommandArgs":["apply","-auto-approve","-input=false"],"level":"info","msg":"","time":"2024-01-16T15:53:35Z","workingDir":"/projects/gruntwork/terragrunt-tests/run-all/app3"}
{"executedCommandArgs":["apply","-auto-approve","-input=false"],"level":"info","msg":"No changes. Your infrastructure matches the configuration.","time":"2024-01-16T15:53:35Z","workingDir":"/projects/gruntwork/terragrunt-tests/run-all/app3"}
```

https://gist.github.com/denis256/23a253bf29ebf2956d162d160d4831e2

**Review by brikis98 — APPROVED** [review]
LGTM! This is good to merge and release. Thank you!

**Pull request merged**